tests/src/test/scala/whisk/core/database/test/DatabaseScriptTestUtils.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.database.test
+
+import scala.concurrent.duration.DurationInt
+import scala.io.Source
+
+import org.scalatest.Matchers
+import org.scalatest.concurrent.ScalaFutures
+
+import akka.actor.ActorSystem
+import common.WaitFor
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import spray.json.JsObject
+import spray.json.pimpString
+import whisk.common.Logging
+import whisk.core.WhiskConfig
+import whisk.core.database.CouchDbRestClient
+import whisk.utils.retry
+import common.WhiskProperties
+
+trait DatabaseScriptTestUtils extends ScalaFutures
+    with Matchers
+    with WaitFor {
+
+    implicit val testConfig = PatienceConfig(10.seconds)
+
+    val python = WhiskProperties.python
+
+    val dbProtocol = WhiskProperties.getProperty(WhiskConfig.dbProtocol)
+    val dbHost = WhiskProperties.getProperty(WhiskConfig.dbHost)
+    val dbPort = WhiskProperties.getProperty(WhiskConfig.dbPort)
+    val dbUsername = WhiskProperties.getProperty(WhiskConfig.dbUsername)
+    val dbPassword = WhiskProperties.getProperty(WhiskConfig.dbPassword)
+    val dbPrefix = WhiskProperties.getProperty(WhiskConfig.dbPrefix)
+    val dbUrl = s"${dbProtocol}://${dbUsername}:${dbPassword}@${dbHost}:${dbPort}"
+
+    /** Creates a new database with the given name */
+    def createDatabase(name: String, designDocPath: Option[String])(implicit as: ActorSystem, logging: Logging) = {
+        // Implicitly remove database for sanitization purposes
+        removeDatabase(name, true)
+
+        println(s"Creating database: $name")
+        val db = new ExtendedCouchDbRestClient(dbProtocol, dbHost, dbPort.toInt, dbUsername, dbPassword, name)
+        retry({ db.createDb().futureValue shouldBe 'right }, N = 10, waitBeforeRetry = Some(500.milliseconds))
+
+        retry({
+            val list = db.dbs().futureValue.right.get
+            list should contain(name)
+        }, N = 10, waitBeforeRetry = Some(500.milliseconds))
```

Comment:
Would it be beneficial to create a local alias for retry, which is curried with the retry parameters? Like

```
def retry(task: => Any) = whisk.utils.retry(task, N = 10, waitBeforeRetry = Some(500.milliseconds))
```

(Don't know the type of `task` but you get the idea)